Makefile

```diff
@@ -0,0 +1,120 @@
+# ======================================================================
+# 
+#  Setup
+# 
+# ======================================================================
+.DEFAULT_GOAL := help
+
+# Dotenv load
+VARS_OLD := $(.VARIABLES)
+DOTENV_FILE ?= ./.env
+ENV_FILE ?= $(DOTENV_FILE)
+ifneq (,$(wildcard $(ENV_FILE)))
+	include $(ENV_FILE)
+	export $(shell sed 's/=.*//' $(ENV_FILE))
+endif
+
+# ======================================================================
+# 
+#  Standard commands
+# 
+# ======================================================================
+aerospike: $(DOCKER)  ## Start aerospike docker instance
+	mkdir -p $(AEROSPIKE_DATA_PATH)
+	$(DOCKER) run --rm -it $(AEROSPIKE_DOCKER_RUN) --name aerospike $(AEROSPIKE_URL)
+
+audit: $(CARGO) $(AUDIT)  ## Check for security vulnerabilities
+	$(CARGO) audit
+
+benchmark: $(CARGO)  ## Benchmarks code
+	$(CARGO) bench
+
+build: $(CARGO)  ## Compiles crate workspace
+	$(CARGO) build --release --workspace
+
+doc: $(CARGO)  ## Build documentation
+	$(CARGO) doc --no-deps
+
+help: ## Show this menu 
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+lint: $(CARGO) clippy  ## Run style check
+	$(CARGO) clippy --all --all-features -- -D warnings
+
+release: $(CARGO) | style lint doc build test benchmark audit  ## Run everything needed for a release
+
+style: | $(CARGO) rustfmt ## Run style check
+	$(CARGO) fmt --all -- --check
+
+test: $(CARGO)  ## Run tests
+	$(CARGO) test --release
+
+# ======================================================================
+# 
+#   Support
+# 
+# ======================================================================
+# docker
+DOCKER := $(shell which docker || echo /usr/local/bin/docker)
+
+$(DOCKER):
+	@echo "`docker` must be installed and available from PATH"
+	@env | grep -i ^PATH
+	@exit 1
+
+# aerospike
+AEROSPIKE_DATA_PATH ?= $(HOME)/.aerospike/data
+AEROSPIKE_DEFAULT_TTL ?= 0
+AEROSPIKE_DOCKER_REPO ?= aerospike/aerospike-server
+AEROSPIKE_MEMORY ?= 1
+AEROSPIKE_NAMESPACE ?= test
+AEROSPIKE_STORAGE ?= 8
+AEROSPIKE_URL ?= $(AEROSPIKE_DOCKER_REPO):$(AEROSPIKE_VERSION)
+AEROSPIKE_VERSION ?= 5.4.0.2
```

Comment:
I think `latest` would be a better default, rather than hard-coding a specific version number.

Reply:
@jhecking

I would like to also use `latest`, because I think that's a set it and forget it type of thing.

How do we handle the problem where `latest` has pulled from dockerhub months ago and is pointing to a (possibly) much older version?  I actually had this issue last week because on my localhost, the `latest` wasn't pointing to version 5.4 version and I don't think docker actually checks datetime timestamps and/or image hashes to see if there's an update to the `latest` tag.

I think for CI/CD, `latest` would work just fine, though.

Reply:
You could add `$(DOCKER) pull $(AEROSPIKE_URL)` to the `aerospike` target to ensure you are always running the latest version.